soh/src/code/z_map_mark.c

```diff
@@ -94,122 +95,92 @@ void MapMark_DrawForDungeon(PlayState* play) {
 
     interfaceCtx = &play->interfaceCtx;
 
-    if ((gMapData != NULL) && (play->interfaceCtx.mapRoomNum >= gMapData->dgnMinimapCount[dungeon])) {
```

Comment:
These changes are to rework this code to match the different way map marks are represented in SceneDB. In vanilla, each room has a list of marks that end in a stop marker. This way is horrible wasteful. In SceneDB, there is a list of chest marks, and a list of boss marks. Because drawing each mark is identical except for the texture, I have combined the loops and hoisted some common code out of them to make them more efficient.